app/src/test/java/com/github/swent/echo/compose/components/ListDrawerTest.kt

```diff
@@ -0,0 +1,116 @@
+package com.github.swent.echo.compose.components
+
+import androidx.compose.ui.test.assertTextEquals
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.swent.echo.data.SAMPLE_EVENTS
+import java.time.format.DateTimeFormatter
+import kotlin.math.min
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class ListDrawerTest {
+    @get:Rule val composeTestRule = createComposeRule()
+
+    @Before
+    fun setUp() {
+        composeTestRule.setContent { ListDrawer(SAMPLE_EVENTS) }
+    }
+
+    @Test
+    fun testListDrawerExists() {
+        composeTestRule.onNodeWithTag("list_drawer").assertExists()
+    }
+
+    @Test
+    fun testAllEventsDisplayedCorrectly() {
+        // Check max 5 events to avoid interminable test when we will have a lot of events
+        for (i in 0..min(SAMPLE_EVENTS.size - 1, 4)) {
+            composeTestRule
+                .onNodeWithTag("list_event_item_${SAMPLE_EVENTS[i].eventId}")
+                .assertExists()
+            composeTestRule
```

Comment:
Knitpick: It could be good practice to wrap the duplicate code in the loop into a function to avoid code duplication and a cleaner code.

Reply:
I'm not sure to see which duplication you are talking about ?

Reply:
I meant wrapping the code inside the for loop in a separate function and then call the following function while testing for 5 sample events. In this way, if you would ever want to modify the tests, it would be easier to do when the logic is wrapped in a function.